src/js/ripple/log.js

```diff
@@ -61,17 +62,36 @@ Log.prototype.error = Log.makeLevel(4);
  * implementations. This is the logging engine used in Node.js.
  */
 var BasicLogEngine = {
-  logObject: function logObject(msg) {
-    var args = Array.prototype.slice.call(arguments, 1);
-
-    args = args.map(function(arg) {
-      return JSON.stringify(arg, null, 2);
+  logObject: function logObject(level, message) {
+    var timestamp = '[' + new Date().toISOString() + ']';
+    var location = (new Error()).stack.split('\n')[4].replace(/^\s+/, '');
+
+    var args = [
+      timestamp,
+      message,
+      '--',
+      location,
+      '\n'
+    ];
+
+    Array.prototype.slice.call(arguments, 2).forEach(function(arg) {
+      args.push(JSON.stringify(arg, null, 2));
     });
 
-    args.unshift(msg);
-    args.unshift('[' + new Date().toISOString() + ']');
+    var logFn;
+
+    switch (level) {
+      case 3:
+        logFn = console.warn;
+        break;
+      case 4:
+        logFn = console.error;
+        break;
+      default:
+        logFn = console.log;
+    }
```

Comment:
This could be simplified if makeLevel took the log function itself (console.warn, console.log, or console.error) rather than an integer which then needs to be mapped back to a function here.